client/fingerprint/consul.go

```diff
@@ -173,11 +183,54 @@ func (f *ConsulFingerprint) connect(info agentconsul.Self) (string, bool) {
 	return strconv.FormatBool(c), ok
 }
 
-func (f *ConsulFingerprint) grpc(info agentconsul.Self) (string, bool) {
+func (f *ConsulFingerprint) grpc(scheme string) func(info agentconsul.Self) (string, bool) {
+	return func(info agentconsul.Self) (string, bool) {
+
+		// The version is needed in order to understand which config object to
+		// query. This is because Consul 1.14.0 added a new gRPC port which
+		// broke the previous behaviour.
+		v, ok := info["Config"]["Version"].(string)
+		if !ok {
+			return "", false
+		}
+
+		consulVersion, err := version.NewVersion(v)
+		if err != nil {
+			return "", false
+		}
+
+		// This check ensures that users running a 1.14.0 pre-release get the
+		// desired behaviour, otherwise 1.14.0-beta1 < 1.14.0 will be truthy.
+		versionsMatch := slices.Equal(consulGRPCPortChangeVersion.Segments(), consulVersion.Segments())
```

Comment:
We probably don't use it everywhere we should be but just FYI [`Version.Core`](https://pkg.go.dev/github.com/hashicorp/go-version?utm_source=godoc#Version.Core) lets you extract a version without `-beta1`, etc.

Reply:
Thanks that is a lot cleaner. The update can be seen within commit: https://github.com/hashicorp/nomad/pull/15309/commits/cebd6b59ed3afbc521678cb3441be1c2ce5b64d2